### PR TITLE
Make  'Cluster points within a layer' example code compatible with GL JS / iOS

### PR DESCRIPTION
**Pull request by OttyLab**
### Summary of changes

Improve the circle layer cluster example so that it becomes compatible with [GL JS](https://docs.mapbox.com/mapbox-gl-js/example/cluster/) / [iOS](https://github.com/mapbox/mapbox-maps-ios/blob/v10.10.1/Apps/Examples/Examples/All%20Examples/Annotations/SymbolClusteringExample.swift) example codes

GL JS and iOS examples uses `stop` expression to change the circle color. Therefore, this PR uses same technique.

The code change does not affect the behavior like;

before:
main_s

after:
fix_s

### User impact (optional)

This update helps users understand how to implement same stuff on different environment.

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [x] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [ ] Write tests for all new functionality. If tests were not written, please explain why. --> it's an example code
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc). --> N/A
 - [x] Add example if relevant.
 - [ ] Document any changes to public APIs. --> NA
 - [ ] Run `make update-api` to update generated api files, if there's public API changes, otherwise the `verify-api-*` CI steps might fail. --> NA
 - [x] Update [CHANGELOG.md](../CHANGELOG.md) or use the label 'skip changelog', otherwise `check changelog` CI step will fail.
 - [ ] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` firstly and then port to `v10.[version]` release branch. --> NA

Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).

### CODE REVIEW

**Comment by pengdev** [comment]
@OttyLab thanks for the PR, please sign the CLA bot, and check the failing verify-code CI and make sure `./gradlew ktlint` succeeds.

File: app/src/main/java/com/mapbox/maps/testapp/examples/CircleLayerClusteringActivity.kt [hunk 1]
```diff
@@ -140,47 +141,31 @@ class CircleLayerClusteringActivity : AppCompatActivity() {
...
+              literal(layers[0][0].toDouble())
+              literal ( ColorUtils.colorToRgbaString(layers[0][1]) )
+            }
+          }
+        )
+        circleRadius(18.0)
+        filter(
+          has {
+            literal("point_count")
+          }
```
Comment (pengdev) [inline app/src/main/java/com/mapbox/maps/testapp/examples/CircleLayerClusteringActivity.kt:165]: Nit: can be simplified to 
```suggestion
          has("point_count")
```

File: app/src/main/java/com/mapbox/maps/testapp/examples/CircleLayerClusteringActivity.kt [hunk 2]
```diff
@@ -140,47 +141,31 @@ class CircleLayerClusteringActivity : AppCompatActivity() {
...
-                  }
-                  literal(layers[i - 1][0].toLong())
-                }
-              }
+    // Add clusters' circles
+    style.addLayer(
+      circleLayer("clusters", GEOJSON_SOURCE_ID) {
+        circleColor(
+          step{
+            get {literal("point_count")}
```
[the remark this answers is not in the document]
Comment (OttyLab) [reply app/src/main/java/com/mapbox/maps/testapp/examples/CircleLayerClusteringActivity.kt]: @pengdev 

Thanks, fixed

File: app/src/main/java/com/mapbox/maps/testapp/examples/CircleLayerClusteringActivity.kt (hunk 1 above)
Comment (OttyLab) [reply app/src/main/java/com/mapbox/maps/testapp/examples/CircleLayerClusteringActivity.kt]: @pengdev 

Thanks, fixed

**Review by pengdev — APPROVED** [review]

**Review by ank27 — APPROVED** [review]

**Review by yunikkk — APPROVED** [review]

**Pull request merged**